modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,441 @@
+        struct MVKernel
+        {
+            uint8_t *image1;
+            uint8_t *image2;
+            uint8_t *integralLeft;
+            uint8_t *integralRight;
+            MVKernel(uint8_t *img, uint8_t *img2, uint8_t *integralL, uint8_t *integralR): image1(img),image2(img2),integralLeft(integralL), integralRight(integralR){}
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int &c, int &c2) const
+            {
+                (void)rrWidth;
+                (void)w2;
+                (void)rWidth;
+                (void)jj;
+                (void)j;
+                (void)c;
+                (void)c2;
+            }
+        };
+        //single version of MV
+        struct MVKernelSingle
+        {
+            uint8_t *image1;
+            uint8_t *integralLeft;
+            MVKernelSingle(uint8_t *img, uint8_t *integralL): image1(img),integralLeft(integralL){}
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int &c) const
+            {
+                (void)rrWidth;
+                (void)w2;
+                (void)rWidth;
+                (void)jj;
+                (void)j;
+                (void)c;
+            }
+        };
+        //!kernel that takes the pixels from certain positions from a patch
+        //!offers verry good results
+        struct StarKernel
+        {
+            uint8_t *image1;
+            uint8_t *image2;
+            StarKernel(uint8_t *img, uint8_t *img2): image1(img),image2(img2){}
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int &c, int &c2) const
+            {
+                (void)rrWidth;
+                (void)w2;
+                (void)rWidth;
+                (void)jj;
+                (void)j;
+                (void)c;
+                (void)c2;
+            }
+        };
+        //single version of StarKernel
+        struct StarKernelSingle
+        {
+            uint8_t *image1;
+            StarKernelSingle(uint8_t *img): image1(img){}
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int &c, int &c2) const
+            {
+                (void)rrWidth;
+                (void)w2;
+                (void)rWidth;
+                (void)jj;
+                (void)j;
+                (void)c;
+                (void)c2;
+            }
+        };
+        //!Compares pixels from a patch giving high weights to pixels in which
+        //!the intensity is higher. The other pixels receive a lower weight
+        struct MCTKernel
+        {
+            uint8_t *image1;
+            uint8_t *image2;
+            int t;
+            MCTKernel(uint8_t * img,uint8_t *img2, int threshold) : image1(img),image2(img2), t(threshold) {}
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int &c, int &c2) const
+            {
+                (void)w2;
+                if (image1[rrWidth + jj] > image1[rWidth + j] - t)
```

Comment:
I think we can make the last step and remove the duplication inside such kernels as well. Something like `int c[2]; uint8_t *images[2];` and a for loop from 0 to 1 will do this.